GVFS/GVFS.Common/ProductUpgraderFactory.cs

```diff
@@ -15,6 +16,12 @@ public class ProductUpgraderFactory
             bool dryRun = false,
             bool noVerify = false)
         {
+            newUpgrader = NuGetUpgrader.NuGetUpgrader.Create(tracer, dryRun, noVerify, out error);
+            if (newUpgrader != null)

```

Comment:
Same questions.  Why is the NuGetUpgrader first instead of the GitHubUpgrader?

Reply:
> Why is the NuGetUpgrader

For this iteration - we will prefer to use the NuGetUpgrader if it is "enabled" (that is, if any of the config entries for NuGet Upgrader are populated, then we will assume the user means to use NuGet Upgrader instead of GitHub Upgrader).

As a followup item, we might refine this logic to having the user explicitly opt-in to NuGetUpgrader, possibly via the `upgrade.ring` setting (e.g. you would set upgrade.ring to NuGetUpgrader to get NuGet functionality).

> If Create could possibly return null and also has out string error then should it not be TryCreate

I could - the downside is yet another `out` variable, when we could go with the pattern that it will return a new object, or else return null + error string....